### DISCUSSION TITLE
HTTPS redirection (duplicated?)

### DISCUSSION THREAD

**Discussion by claha** [techno-tim/techno-tim.github.io#169] [Questions] [answered]
Thanks for this repo and all the videos! Currently setting up traefik myself and I'm looking at the http to https redirection. 

To me it seems that a redirection from http to https is defined here (at entrypoints.http) https://github.com/techno-tim/techno-tim.github.io/blob/master/reference_files/traefik-portainer-ssl/traefik/traefik.yml
I use this myself and it seems to work.

But it is also defined as a middleware here https://github.com/techno-tim/techno-tim.github.io/blob/master/reference_files/traefik-portainer-ssl/traefik/docker-compose.yml

Are both needed? If not, is there a preferred way?

I guess a third option is to instead use the https-redirectscheme middleware defined in https://github.com/techno-tim/techno-tim.github.io/blob/master/reference_files/traefik-portainer-ssl/traefik/config.yml and then instead use `middleware=https-redirectscheme@file` in docker-compose.

**Comment by timothystewart6** [comment, marked as answer]
Hey!  This might be duplicated but I really didn't know how people were going to use it.  You can try with whatever works for you!

**Comment by claha** [comment]
I guess that is the beauty of homelabbing, you can do what works for you :)

I tried reading up a bit more on the documentation and if I understood it correctly. Adding the redirection in traefik.yml will cause a permanent redirection from http to https for all services, i.e. no service will be available over http. This could potentially be a problem if you have a service that can't be accessed over https, I will see along my journey if I run into this problem.

https://doc.traefik.io/traefik/routing/entrypoints/#redirection
